mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -255,18 +256,37 @@ func (sp *StateProcessor[
 		)
 	}
 
+	// Ensure the EVM inflation withdrawal is the first withdrawal.
+	if numWithdrawals == 0 {
+		return ErrNoWithdrawals
+	}
+	if !expectedWithdrawals[0].GetAddress().Equals(
+		sp.cs.EVMInflationAddress(),
+	) ||
+		expectedWithdrawals[0].GetAmount() != math.Gwei(
+			sp.cs.EVMInflationPerBlock(),
+		) {
+		return ErrFirstWithdrawalNotEVMInflation
+	}
```

Comment:
nit:  why don't we just use equals here? Looks like we build expected payload setting validator index to zero, but we do let withdrawals in block to set the validator differently. Granted we don't really care, I'd rather Expected and actual withdrawal to fully match.